src/object/schema_manager.c

```diff
@@ -15797,13 +15785,23 @@ sm_truncate_class_internal (MOP class_mop)
 	  continue;
 	}
 
-      if (keep_pk == true && c->type == SM_CONSTRAINT_PRIMARY_KEY)
+      if ((c->type == SM_CONSTRAINT_PRIMARY_KEY && classobj_is_pk_referred (class_mop, c->fk_info, false, NULL))
+	  || !sm_is_possible_to_recreate_constraint (class_mop, class_, c))
 	{
-	  /* Do not save PK referred by FK. the PK can't be dropped. */
-	  continue;
+	  /*
+	   * 1. the PK referred to by a FK.
```

Comment:
Please explain more detail about this condition and leave it to related Jira issue. Because it is thought to be the most commonly used case. (I remember you shared the reason with me. But, I ask you once again)

Reply:
Ok, I've added an overview of these undroppable indexes on the CBRD-23931. When it comes to details, do you mean the description of CBRD-23938 is not enough? To give a precise reason for this need more code analysis.

Reply:
It's enough. (CBRD-23931)